src/connectors/numeric-menu/connectNumericMenu.js

```diff
@@ -170,80 +170,67 @@ export default function connectNumericMenu(renderFn, unmountFn = noop) {
+        const values = searchParameters.getNumericRefinements(attribute);
+
+        const equal = values['='] && values['='][0];
+
         if (equal || equal === 0) {
           return {
             ...uiState,
             numericMenu: {
               ...uiState.numericMenu,
-              [attribute]: `${currentRefinements['=']}`,
+              [attribute]: `${values['=']}`,
             },
           };
         }
 
-        const lowerBound =
-          (currentRefinements['>='] && currentRefinements['>='][0]) || '';
-        const upperBound =
-          (currentRefinements['<='] && currentRefinements['<='][0]) || '';
-
-        if (lowerBound !== '' || upperBound !== '') {
-          if (
-            uiState.numericMenu &&
-            uiState.numericMenu[attribute] === `${lowerBound}:${upperBound}`
-          )
-            return uiState;
-          return {
-            ...uiState,
-            numericMenu: {
-              ...uiState.numericMenu,
-              [attribute]: `${lowerBound}:${upperBound}`,
-            },
-          };
+        const min = (values['>='] && values['>='][0]) || '';
+        const max = (values['<='] && values['<='][0]) || '';
+
+        if (min === '' && max === '') {
+          return uiState;
         }
 
-        return uiState;
+        return {
+          ...uiState,
+          numericMenu: {
+            ...uiState.numericMenu,
+            [attribute]: `${min}:${max}`,
+          },
+        };
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        let clearedParams = searchParameters.clearRefinements(attribute);
         const value = uiState.numericMenu && uiState.numericMenu[attribute];
 
+        const withoutRefinements = searchParameters.clearRefinements(attribute);
+
         if (!value) {
-          return clearedParams;
+          return withoutRefinements.setQueryParameters({
+            numericRefinements: {
+              ...withoutRefinements.numericRefinements,
+              [attribute]: {},
+            },
+          });
         }
 
-        const valueAsEqual = value.indexOf(':') === -1 && value;
+        const isExact = value.indexOf(':') === -1;
 
-        if (valueAsEqual) {
-          return clearedParams.addNumericRefinement(
-            attribute,
-            '=',
-            valueAsEqual
-          );
+        if (isExact) {
+          return withoutRefinements.addNumericRefinement(attribute, '=', value);
         }
 
-        const [lowerBound, upperBound] = value.split(':').map(parseFloat);
+        const [min, max] = value.split(':').map(parseFloat);
 
-        if (isFiniteNumber(lowerBound)) {
-          clearedParams = clearedParams.addNumericRefinement(
-            attribute,
-            '>=',
-            lowerBound
-          );
-        }
+        const withMinRefinement = isFiniteNumber(min)
+          ? withoutRefinements.addNumericRefinement(attribute, '>=', min)
+          : withoutRefinements;
```

Comment:
Here the value won't have the default value but it should be. We have to perform all the checks at the top to ensure that if any of the value is incorrect we return the default value. We can implement the two fix at the same time since they are related.

Reply:
In a separate PR then?

Reply:
Yes.